File: Data Structures & Algorithms/longest-consecutive-sequence/submission-6.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int longestLength = 0;

        Map<Integer, Boolean> exploredMap = new HashMap<>();

        for(int num: nums)
        {
            exploredMap.put(num, Boolean.FALSE);
        }


        for(int num: nums)
        {
            int currentLength = 1;
            
            //move forward
            int nextNum = num + 1;
            while(exploredMap.containsKey(nextNum) && exploredMap.get(nextNum) == false)
            {
                currentLength += 1;
                exploredMap.put(nextNum, Boolean.TRUE);
                nextNum += 1;
            }
            
            //move previous
            int prevNum = num - 1;

            while(exploredMap.containsKey(prevNum) && exploredMap.get(prevNum) == false)
            {
                currentLength += 1;
                exploredMap.put(prevNum, Boolean.TRUE);
                prevNum -= 1;
            }

            longestLength = Math.max(longestLength, currentLength);

            
        }
            return longestLength;
    }
}

File: Data Structures & Algorithms/maximum-subarray/submission-6.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        
        total_sum:int = 0

        maxLength: int = nums[0];
        for i in range(0,len(nums)):
            
            total_sum = total_sum + nums[i]

            maxLength = max(maxLength,total_sum)


            if total_sum < 0:
                total_sum = 0;

        return maxLength